osu.Game.Rulesets.Taiko/Mods/TaikoModHidden.cs

```diff
@@ -1,23 +1,82 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Game.Configuration;
 using osu.Game.Rulesets.Mods;
 using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.Taiko.Objects;
+using osu.Game.Rulesets.Taiko.Objects.Drawables;
+using osu.Game.Rulesets.Taiko.UI;
+using osu.Game.Rulesets.Timing;
+using osu.Game.Rulesets.UI;
 
 namespace osu.Game.Rulesets.Taiko.Mods
 {
-    public class TaikoModHidden : ModHidden
+    public class TaikoModHidden : ModHidden, IApplicableToDrawableRuleset<TaikoHitObject>
     {
         public override string Description => @"Beats fade out before you hit them!";
         public override double ScoreMultiplier => 1.06;
-        public override bool HasImplementation => false;
+        private const double relative_fade_position = 0.5;
+        private DrawableTaikoRuleset ruleset;
+        private readonly MultiplierControlPoint searchPoint = new MultiplierControlPoint();
+
+        [SettingSource("Hide Time", "Controls the time the note stays hidden")]
+        public BindableNumber<double> VisibilityMod { get; } = new BindableDouble
+        {
+            MinValue = 0.35,
+            // Max visibility is only to be used with max playfield size
+            MaxValue = 1.35,
```

Comment:
Need explanation for this. What happens if it isn't used with max playfield size?

Reply:
If it isn't used with max playfield size, no notes are visible. Should it be scaled with playfield size?

Reply:
Not sure, but if it's coupled somehow to the other mod, then the ideal state of things would be that one mod's settings influence the other's allowable bounds to prevent brokenness. However this is unprecedented in lazer and we don't really have a good way of doing that sort of stuff.

Incidentally taiko HD and HR also seem to be somehow intertwined on stable, [which already came up once](https://github.com/ppy/osu/pull/11364#discussion_r550181930) but ended up being unresolved for now.

Reply:
removed this mod for now